mixnet/client/src/config.rs

```diff
@@ -21,10 +24,12 @@ pub enum MixnetClientMode {
 
 impl MixnetClientMode {
     pub(crate) async fn run(&self) -> Result<MessageStream, MixnetClientError> {
+        let ack_cache = Arc::new(Mutex::new(HashMap::new()));
```

Comment:
I haven't read all yet. Is this ACK something like SURB in Nym? As I noted in https://github.com/logos-co/nomos-node/issues/322#issuecomment-1709375953, I thought we don't need SURB yet because SURB generates at least 2x network bandwidth, which is what we're not currently sure if it's okay to accept. If this ACK is different from SURB, or if we really need this ACK, we may have to sacrifice downsides. I'll review this in detail soon.

Reply:
IMO, the ACK here is not the same as SURB, and please correct me if I am wrong. Indeed, ACK here will lead to an extra kind of message. I add ack here because I cannot find another way to confirm if such a message from the sender has been received by its receiver.

Reply:
I do not think we really need to confirm the message arrived. It is mostly that the message **was sent**. So, if the outgoing connection failed for whatever reason we may try again. Or at least that was my understanding for the problem. @youngjoon-lee can confirm.

Reply:
Yes, exactly. We don't really need the delivery guarantee (to the final destination) at this stage. As a message sender, the thing we can easily do right now (by retrying) is making sure that the packet is at least delivered to the "first" mixnode successfully. I think that's enough for now.